src/runtime/VMInstance.cpp

```diff
@@ -133,6 +133,17 @@ void VMInstance::compressStringsIfNeeds(uint64_t currentTickCount)
 }
 #endif
 
+static void markHashSet(GC_word* desc, size_t base)
+{
+#if defined(COMPILER_MSVC) || defined(COMPILER_CLANG_CL)
+    GC_set_bit(desc, base + 2); // m_ht.m_buckets_data
+    GC_set_bit(desc, base + 5); // m_ht.m_buckets
+#else
+    GC_set_bit(desc, base + 1); // m_ht.m_buckets_data
+    GC_set_bit(desc, base + 4); // m_ht.m_buckets
+#endif
```

Comment:
One more question, why are these offsets different?
Cant't we just simply present it like `GC_WORD_OFFSET(HashSet, m_ht.m_buckets_data)`?

Reply:
we cannot access m_ht.m_buckets_data here since the m_ht variable is private